src/FastState.cpp

```diff
@@ -90,9 +94,16 @@ void FastState::play_move(int vertex) {
 
 void FastState::play_move(int color, int vertex) {
     if (vertex != FastBoard::PASS) {
+        board.m_hash ^= Zobrist::zobrist_ko[m_komove];
         int kosq = board.update_board(color, vertex);
+        if ( board.get_square(kosq) == FastBoard::EMPTY &&
+            !board.is_suicide(kosq, !color)) {
```

Comment:
Nice fix, but maybe this logic belongs in update_board? It can be its responsibility to not return fake ko squares.